plugins/plugin-debugger/che-plugin-debugger-ide/src/main/java/org/eclipse/che/plugin/debugger/ide/debug/AbstractDebugger.java

```diff
@@ -582,6 +590,29 @@ public void stepOver() {
   }
 
   @Override
+  public void runToLocation(Location location) {
```

Comment:
I've reconsidered a lot. There is a pitfall here if location isn't executable.
We can get rid of all server side changes and support RunToCursor action for all debugger implementation.

Reply:
Add breakpoint, if promise is succeeded then resume debugger and instantiate some class as Debugger observer. Once application is resumed then remove breakpoint and remove that class as a observer of the Debugger.

Reply:
done